### PR TITLE
Enable generation for ES6 modules

### LINKED ISSUE #36: Unable to generate typings for ES6 modules
The `typescriptModule` parameter is required, but ES6 modules do not rely on the module declaration in the generated typings file.

```
module MyApp.Http { // do not need this

    export interface HttpEndpointOptions {
        serviceIdentifier?: string;
        endpointPath: string;
        endpointName: string;
        method: string;
        mediaType: string;
        requiredHeaders: string[];
        pathArguments: any[];
        queryArguments: any;
        data?: any;
    }

    export interface HttpApiBridge {
        callEndpoint<T>(parameters: HttpEndpointOptions): Promise<T>;
    }
}
```

This way I can refer to this file as follows:

```
import { HttpApiBridge } from "./generated/httpApiBridge";
```

### PR DESCRIPTION
**Pull request by rcchen**
By omitting the `typescriptModule` parameter, it is possible to generate typings for use in module-based projects.

Fixes #36

### CODE REVIEW

**Comment by rcchen** [comment]
@ryan10132 let's iterate on this one, it works but has two minor issues and a lack of tests right now

**Comment by ryan10132** [comment]
Cool, thanks @rcchen I think I know the issue, I'll add a commit to yours to illustrate (then we can squash it in to your first). I also think we should add an option something like "generateES6" that you set to true and defaults to false, for sake of clarity.

**Comment by ryan10132** [comment]
Well I can't push to your branch (doh) but here's what you're looking for I believe
```
diff --git a/typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceEmitter.java b/typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceEmitter.java
index d084c8a..4ef2ff6 100644
--- a/typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceEmitter.java
+++ b/typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceEmitter.java
@@ -76,7 +76,11 @@ public final class ServiceEmitter {
         ByteArrayOutputStream typeDeclarations = new ByteArrayOutputStream();
         Type[] types = new Type[referencedClasses.size()];
         referencedClasses.toArray(types);
-        typescriptGenerator.generateEmbeddableTypeScript(Input.from(types), Output.to(typeDeclarations), true, 1);
+        int indentationLevel = 1;
+        if (settings.typescriptModule().isPresent()) {
+            indentationLevel = 0;
+        }
+        typescriptGenerator.generateEmbeddableTypeScript(Input.from(types), Output.to(typeDeclarations), true, indentationLevel);
         writer.write(new String(typeDeclarations.toByteArray()));
     }
```

**Comment by rcchen** [comment]
Aite let me give that a shot and push an update up. So we're using an explicit flag instead of `typescriptModule = null`?

**Comment by ryan10132** [comment]
Yea I think that will add some clarity. I'm thinking that if you don't have a module, then you don't add the module declaration and the indenation.

If you set es6 to true, then you also import from the httpApiBridge file the es6 way

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceGenerator.java [hunk 1]
```diff
@@ -72,6 +72,14 @@ public void generateTypescriptService(Class<?> serviceClass, List<Type> addition
 
         ServiceModel serviceModel = new ServiceClassParser().parseServiceClass(serviceClass, settings, serviceClassesToMerge);
         ServiceEmitter serviceEndpointEmitter = new ServiceEmitter(serviceModel, settings, writer);
+        
+        if (!settings.typescriptModule().isPresent()) {
```
Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceGenerator.java]: I'm suggesting changing this to be settings.isES6()

**Comment by ryan10132** [comment]
I can add a test for this after you merge this btw

**Comment by rcchen** [comment]
Awesome, I should have something up for you in five min or so

**Comment by rcchen** [comment]
Indentation is still off

```
// A potential copyright header
// A desired generated message
import { IHttpEndpointOptions, IHttpApiBridge } from "./httpApiBridge.ts";

    export interface IMyObject {
        payload: string;
    }

    export interface IPlanet {
        name: string;
        radiusKm: number;
    }

export interface IMyService {
    helloWorld(): ng.IPromise<IMyObject>;
    planets(): ng.IPromise<IPlanet[]>;
```

but the rest works now. I have an `emitES6` flag that gets a boolean passed

**Comment by rcchen** [comment]
Ah found it, should be a not. Pushing an update and should be good to go.

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/TypescriptServiceGeneratorConfiguration.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public DuplicateMethodNameResolver duplicateEndpointNameResolver() {
 
     /**
      * If true, emit method signatures even when it would create duplicate java names (results in typescript that does not compile).
-     * If faslse, emit a warning instead.
+     * If false, emit a warning instead.
```
Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/TypescriptServiceGeneratorConfiguration.java]: :tada:

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceGenerator.java [hunk 3]
```diff
@@ -72,6 +72,14 @@ public void generateTypescriptService(Class<?> serviceClass, List<Type> addition
 
         ServiceModel serviceModel = new ServiceClassParser().parseServiceClass(serviceClass, settings, serviceClassesToMerge);
         ServiceEmitter serviceEndpointEmitter = new ServiceEmitter(serviceModel, settings, writer);
+        
+        if (settings.emitES6()) {
```
Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceGenerator.java]: I'm curious, have you tested that this compiles in your setup at all?

**Comment by ryan10132** [comment]
Have you tested this in any way? Like generated the services you needed, copied into your code base and seen that they compiled?

**Comment by rcchen** [comment]
I ran `MyServiceGenerator` in `typescript-service-generator-examples` with the flag set and it worked as expected. Haven't tested it with my code base but I'm reasonably sure it should work since the top level structures are identical (and those are the ones that this change directly impacts)

**Comment by ryan10132** [comment]
Cool, is this good with you to merge?

**Comment by rcchen** [comment]
Yeah let's do it :balloon:

**Comment by rcchen** [comment]
Sweet, can you cut a release so I can try it out tomorrow?

**Comment by ryan10132** [comment]
done

**Pull request merged**